core-authentication/src/main/java/gg/solarmc/loader/authentication/UnauthenticatedUser.java

```diff
@@ -19,51 +19,43 @@
 
 package gg.solarmc.loader.authentication;
 
-import java.util.Objects;
-import java.util.UUID;
-
-public final class UnauthenticatedUser {
-
-	private final UUID mcUuid;
-	private final String username;
+import gg.solarmc.loader.Transaction;
 
-	public UnauthenticatedUser(UUID mcUuid, String username) {
-		this.mcUuid = Objects.requireNonNull(mcUuid);
```

Comment:
This isnt really a proper review but it caught my eye - Shouldn't we use some kind of standardized @Nullable annotation instead of adding a line every time we want to verify a variable for nullity?

Reply:
Nullability annotations are not enforced at runtime. Some of them (with Retention(RUNTIME)) are enforceable if you use AOP and write a brief runtime nullness checker, but I do not think we need to do that. Objects.requireNonNull is perfectly suitable